src/python/graplctl/graplctl/cli.py

```diff
@@ -0,0 +1,612 @@
+import dataclasses
+import os
+import uuid
+
+from typing import List
+
+from mypy_boto3_ec2 import EC2ServiceResource
+import mypy_boto3_ec2.service_resource as ec2_resources
+from mypy_boto3_cloudwatch.client import CloudWatchClient
+from mypy_boto3_cloudwatch.type_defs import MetricTypeDef
+from mypy_boto3_sns.client import SNSClient
+from mypy_boto3_ssm import SSMClient
+from mypy_boto3_route53 import Route53Client
+
+import boto3
+import click
+
+from . import __version__
+from . import dgraph_ops
+from . import docker_swarm_ops
+
+SESSION = boto3.Session(profile_name=os.getenv("AWS_PROFILE", "default"))
+
+EC2: EC2ServiceResource = SESSION.resource("ec2", region_name=os.getenv("AWS_REGION"))
+SSM: SSMClient = SESSION.client("ssm")
+CLOUDWATCH: CloudWatchClient = SESSION.client(
+    "cloudwatch", region_name=os.getenv("AWS_REGION")
+)
+SNS: SNSClient = SESSION.client("sns", region_name=os.getenv("AWS_REGION"))
+ROUTE53: Route53Client = SESSION.client("route53", region_name=os.getenv("AWS_REGION"))
+
+#
+# main entrypoint for grapctl
+#
+
+
+@dataclasses.dataclass
+class GraplctlState:
+    grapl_region: str
+    grapl_prefix: str
+    grapl_version: str
+
+
+@click.group()
+@click.version_option(version=__version__)
+@click.option(
+    "-r",
+    "--grapl-region",
+    type=click.Choice(docker_swarm_ops.REGION_TO_AMI_ID.keys()),
+    envvar="GRAPL_REGION",
+    help="grapl region to target [$GRAPL_REGION]",
+)
+@click.option(
+    "-p",
+    "--grapl-prefix",
+    type=click.STRING,
+    envvar="GRAPL_CDK_DEPLOYMENT_NAME",
+    help="grapl deployment name [$GRAPL_CDK_DEPLOYMENT_NAME]",
+)
+@click.option(
+    "-g",
+    "--grapl-version",
+    type=click.STRING,
+    envvar="GRAPL_VERSION",
+    help="grapl version [$GRAPL_VERSION]",
+)
+@click.pass_context
+def main(
+    ctx: click.Context, grapl_region: str, grapl_prefix: str, grapl_version: str
+) -> None:
+    ctx.obj = GraplctlState(grapl_region, grapl_prefix, grapl_version)
+
+
+#
+# swarm operational commands
+#
+
+
+@main.group(help="commands for operating docker swarm clusters")
+def swarm():
+    pass
+
+
+@swarm.command(help="start EC2 instances and join them as a docker swarm cluster")
+@click.option(
+    "-m",
+    "--num-managers",
+    type=click.IntRange(min=1, max=None),
+    help="number of manager nodes to create",
+    required=True,
+)
+@click.option(
+    "-w",
+    "--num-workers",
+    type=click.IntRange(min=1, max=None),
+    help="number of worker nodes to create",
+    required=True,
+)
+@click.option(
+    "-t",
+    "--instance-type",
+    type=click.STRING,
+    help="EC2 instance type for swarm nodes",
+    required=True,
+)
+@click.option(
+    "-i",
+    "--swarm-id",
+    type=click.STRING,
+    help="unique ID for this swarm cluster (random default)",
+    default=str(uuid.uuid4()),
+)
+@click.pass_obj
+def create(
```

Comment:
sorry, here would be where we shoudl validate instance type.
also, thoughts on maybe some sort of preventative mechanism like "you've already `create`'d for this deployment" or something?